cmd/signimage/signimage.go

```diff
@@ -51,38 +51,88 @@ func signFile(filename string, publickey string, privatekey string) error {
 	return nil
 }
 
-func getAuthFromFile(configfile string, repo string) (authn.Authenticator, error) {
+// allSecretFiles  is a comma seperated list of files and directories to search for creds for repo
+// so we need to search through them for the right one.
+// if we get an error we just log it and move on to the next file/dir i
+// and if we find nothing try anonymous auth
+// we're tolerant in our inputs and precise in our inputs. Its the UNIX way!
+func getAuthFromFileList(allSecretFiles string, repo string) authn.Authenticator {
```

Comment:
I looked at WalkDir and I think I rejected it because passing data too and from the WalkDirFunc function is a bit of a pain (iirc you can only do it using global variables) so it makes life easier in some ways and harder in others. 
Plus they are mounted secrets so there should only really be one level in the directory tree so the ability to walk down an arbitrarily deep dirtree isn't providing a huge benefit